vortexasdk/endpoints/canal_transit.py

```diff
@@ -0,0 +1,212 @@
+from datetime import datetime, timedelta
+from typing import Any, Dict, List, Union
+
+from vortexasdk.api import ID
+from vortexasdk.api.shared_types import to_ISODate
+from vortexasdk.endpoints.canal_transit_result import CanalTransitResult
+from vortexasdk.endpoints.endpoints import CANAL_TRANSIT_SEARCH
+from vortexasdk.logger import get_logger
+from vortexasdk.operations import Search
+from vortexasdk.utils import convert_to_list
+
+logger = get_logger(__name__)
+
+
+class CanalTransit(Search):
+    """
+    Canal Transit Endpoint, use this to search through Vortexa's canal transit records.
+
+    A detailed explanation of what Canal Transit is can be found [here](https://docs.vortexa.com/reference/intro-canal-transit).
+    """
+
+    def __init__(self):
+        Search.__init__(self, CANAL_TRANSIT_SEARCH)
+
+    def search(
+        self,
+        filter_vessel_cubic_capacity_min: int = None,
```

Comment:
All of these types need to be set to Optional if they are being initialised as None